rotkehlchen/externalapis/cowswap.py

```diff
@@ -0,0 +1,94 @@
+import json
+import logging
+from typing import TYPE_CHECKING, Any, Final
+
+import requests
+
+from rotkehlchen.db.settings import CachedSettings
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import SupportedBlockchain, Timestamp
+from rotkehlchen.utils.misc import iso8601ts_to_timestamp
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+ORDER_TYPE_CHANGE_TS: Final = Timestamp(1710806400)
+CHAIN_MAPPING: Final = {
+    SupportedBlockchain.ETHEREUM: 'mainnet',
+    SupportedBlockchain.GNOSIS: 'xdai',
+    SupportedBlockchain.ARBITRUM_ONE: 'arbitrum_one',
+}
+
+
+class CowswapAPI:
+    """This is the Cowswap API interface.
+
+    https://api.cow.fi/docs/
+    """
+
+    def __init__(self, database: 'DBHandler', chain: 'SupportedBlockchain') -> None:
+        self.database = database
+        self.session = requests.session()
+
+        if chain not in CHAIN_MAPPING:
+            log.warning(f'Encountered unsupported Cowswap API chain {chain}. Falling back to mainnet.')  # noqa: E501
+            chain = SupportedBlockchain.ETHEREUM
+
+        self.api_url = f'https://api.cow.fi/{CHAIN_MAPPING[chain]}/api/v1'
+
+    def _query(self, endpoint: str) -> dict[str, Any]:
+        """Query cowswap API endpoint."""
```

Comment:
you always need to specify which exceptions a function can raise in its docstring